src/Illuminate/View/Compilers/ComponentTagCompiler.php

```diff
@@ -805,7 +805,7 @@ protected function attributesToStringWithExistingComponentData(array $attributes
             }
 
             $results[] = "'{$attribute}' => \Illuminate\View\Compilers\BladeCompiler::sanitizeComponentAttribute(".(
-                ($attribute === 'attributes') ? $value : ('$componentData[\'data\'][\''.Str::camel($attribute).'\'] ?? null')
+                ($attribute === 'attributes') ? $value : ('$componentData[\'data\'][\''.Str::camel($attribute).'\'] ?? ' . $value)
```

Comment:
The issue here is that any null values will get evaluated twice, which then defeats the purpose of the original PR

Instead, you probably need to use `array_key_exists()` to check if the component data contains that attribute name, and if it does not then use `$value`. If it does contain it, then the existing code works and `?? null` should stay?

Reply:
Something like

```suggestion
                ($attribute === 'attributes') ? $value : (\''.Str::camel($attribute).'\', 'array_key_exists($componentData[\'data\']) ? ($componentData[\'data\'][\''.Str::camel($attribute).'\'] ?? null) : ' . $value)
```

Warning: I have NOT tested this, I'm using the web UI, please check it out locally

Reply:
This seems to work

Reply:
Maybe save the camel case attribute name in a variable so you dont execute it twice?